Library/Homebrew/brew.sh

```diff
@@ -593,52 +605,53 @@ else
   RUBY_DISABLE_OPTIONS="--disable=rubyopt"
 fi
 
-if [[ -z "$HOMEBREW_RUBY_WARNINGS" ]]
+if [[ -z "${HOMEBREW_RUBY_WARNINGS}" ]]
 then
   export HOMEBREW_RUBY_WARNINGS="-W1"
 fi
 
 export HOMEBREW_BREW_DEFAULT_GIT_REMOTE="https://github.com/Homebrew/brew"
-if [[ -z "$HOMEBREW_BREW_GIT_REMOTE" ]]
+if [[ -z "${HOMEBREW_BREW_GIT_REMOTE}" ]]
 then
-  HOMEBREW_BREW_GIT_REMOTE="$HOMEBREW_BREW_DEFAULT_GIT_REMOTE"
+  HOMEBREW_BREW_GIT_REMOTE="${HOMEBREW_BREW_DEFAULT_GIT_REMOTE}"
 fi
 export HOMEBREW_BREW_GIT_REMOTE
 
-if [[ -n "$HOMEBREW_MACOS" ]] || [[ -n "$HOMEBREW_FORCE_HOMEBREW_ON_LINUX" ]]
+if [[ -n "${HOMEBREW_MACOS}" ]] || [[ -n "${HOMEBREW_FORCE_HOMEBREW_ON_LINUX}" ]]
 then
   HOMEBREW_CORE_DEFAULT_GIT_REMOTE="https://github.com/Homebrew/homebrew-core"
 else
   HOMEBREW_CORE_DEFAULT_GIT_REMOTE="https://github.com/Homebrew/linuxbrew-core"
 fi
 export HOMEBREW_CORE_DEFAULT_GIT_REMOTE
 
-if [[ -z "$HOMEBREW_CORE_GIT_REMOTE" ]]
+if [[ -z "${HOMEBREW_CORE_GIT_REMOTE}" ]]
 then
-  HOMEBREW_CORE_GIT_REMOTE="$HOMEBREW_CORE_DEFAULT_GIT_REMOTE"
+  HOMEBREW_CORE_GIT_REMOTE="${HOMEBREW_CORE_DEFAULT_GIT_REMOTE}"
 fi
 export HOMEBREW_CORE_GIT_REMOTE
 
-if [[ -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh" ]]
+if [[ -f "${HOMEBREW_LIBRARY}/Homebrew/cmd/${HOMEBREW_COMMAND}.sh" ]]
 then
-  HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh"
-elif [[ -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh" ]]
+  HOMEBREW_BASH_COMMAND="${HOMEBREW_LIBRARY}/Homebrew/cmd/${HOMEBREW_COMMAND}.sh"
+elif [[ -f "${HOMEBREW_LIBRARY}/Homebrew/dev-cmd/${HOMEBREW_COMMAND}.sh" ]]
 then
-  if [[ -z "$HOMEBREW_DEVELOPER" ]]
+  if [[ -z "${HOMEBREW_DEVELOPER}" ]]
   then
-    git config --file="$HOMEBREW_GIT_CONFIG_FILE" --replace-all homebrew.devcmdrun true 2>/dev/null
+    git config --file="${HOMEBREW_GIT_CONFIG_FILE}" --replace-all homebrew.devcmdrun true 2>/dev/null
     export HOMEBREW_DEV_CMD_RUN="1"
   fi
-  HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh"
+  HOMEBREW_BASH_COMMAND="${HOMEBREW_LIBRARY}/Homebrew/dev-cmd/${HOMEBREW_COMMAND}.sh"
 fi
 
 check-run-command-as-root
 
 check-prefix-is-not-tmpdir
 
-if [[ "$HOMEBREW_PREFIX" = "/usr/local" &&
-      "$HOMEBREW_PREFIX" != "$HOMEBREW_REPOSITORY" &&
-      "$HOMEBREW_CELLAR" = "$HOMEBREW_REPOSITORY/Cellar" ]]
+# shellcheck disable=SC2250
```

Comment:
Not sure the reasoning for this one?

Reply:
avoid checking one variable in the message `Warning: your HOMEBREW_PREFIX~`. 
The following message was displayed when `--enable=all`

```
to $HOMEBREW_CELLAR. Your current HOMEBREW_CELLAR location will stop
   ^--------------^ SC2250: Prefer putting braces around variable references even when not strictly required.
```

Reply:
avoid checking one variable in the message `Warning: your HOMEBREW_PREFIX~`. 
The following message was displayed when `--enable=all`

```
to $HOMEBREW_CELLAR. Your current HOMEBREW_CELLAR location will stop
   ^--------------^ SC2250: Prefer putting braces around variable references even when not strictly required.
```